src/MicrosoftTeams.ts

```diff
@@ -1547,20 +1772,32 @@ namespace microsoftTeams {
   function sendMessageRequest(
     targetWindow: Window,
     actionName: string,
-    // tslint:disable-next-line:no-any
+    // tslint:disable-next-line: no-any
     args?: any[]
   ): number {
     let request = createMessageRequest(actionName, args);
-    let targetOrigin = getTargetOrigin(targetWindow);
-
-    // If the target window isn't closed and we already know its origin, send the message right away; otherwise,
-    // queue the message and send it after the origin is established
-    if (targetWindow && targetOrigin) {
-      targetWindow.postMessage(request, targetOrigin);
+    if (isFramelessWindow) {
+      if (
+        currentWindow.teamsNativeClient &&
```

Comment:
Why can’t you ensure that these injected objects are available from the get-go. This asynchronous initialization seems unnecessary.

Reply:
There is no way on Android to ensure the objects are available from the start. On iOS there is a way, but unfortunately none on Android.

Reply:
This is going to be very problematic since it means you need to handle both the case where the SDK loads first and the case where the injected code load first. Please think through both scenarios and make sure your solution works for them. You might need to set up a poller in the SDK to wait for the injected code to become available or store all the messages in a global queue that the injected code can pump once loaded.

Reply:
Yes, the current code works for both the scenarios. Also, the changes already have the queue approach and all the messages are kept in the queue till the native client is ready. Once the native client is ready, it is currently flushing the queue inside the handleNativeClientInitialize. The poller will work as well, but i guess this is cleaner than that.